### PR TITLE
spelling errors corrected accross various files

### LINKED ISSUE #3645: Running spell-check over the documentation
Fixing the spelling error of the documentation of MDAnalysis by running a spelling check over the files.

### PR DESCRIPTION
**Pull request by aditi2906**
Fixes #3645

Changes made in this Pull Request:
Fixed the typo errors across the documentation of MDAnalysis
 - 

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [ ] Issue raised/referenced?

### CODE REVIEW

File: package/MDAnalysis/analysis/waterdynamics.py [hunk 1]
```diff
@@ -556,7 +556,7 @@ def _sameMolecTandDT(self, selection, t0d, tf):
         select only the particles that are repeated in both frame. This is to
         consider only the molecules that remains in the selection after the dt
         time has elapsed.
-        The result is a list with the indexs of the atoms.
+        The result is a list with the indexes of the atoms.
```
Comment (richardjgowers) [inline package/MDAnalysis/analysis/waterdynamics.py]: Indices

File: package/MDAnalysis/analysis/gnm.py [hunk 2]
```diff
@@ -84,7 +84,7 @@
 .. autofunction:: order_list
 
 .. versionchanged:: 0.16.0
-   removed un-unsed function :func:`backup_file`
+   removed un-unused function :func:`backup_file`
```
Comment (richardjgowers) [inline package/MDAnalysis/analysis/gnm.py]: Unused

**Review by richardjgowers — CHANGES_REQUESTED** [review]
Nice idea but pay attention to details

**Comment by aditi2906** [comment]
@richardjgowers I apologize for the inconvenience caused. Will be vigilant next time onwards :)

File: package/MDAnalysis/analysis/psa.py [hunk 3]
```diff
@@ -190,7 +190,7 @@
 
    .. attribute:: D
 
-      :class:`numpy.ndarray` whichs store calculated distance matrix
+      :class:`numpy.ndarray` which store calculated distance matrix
```
Comment (lilyminium) [inline package/MDAnalysis/analysis/psa.py]:
```suggestion
      :class:`numpy.ndarray` which stores the calculated distance matrix
```

File: package/MDAnalysis/auxiliary/__init__.py [hunk 4]
```diff
@@ -494,16 +494,16 @@
...
 extends :class:`~MDAnalysis.auxiliary.base.AuxReader` by providing the
 following (these may be overwritten by subclasses as appropriate):
 
   ``__init__(auxfile, **kwargs)``
     Open *auxfile* and any additional processing of *kwargs*.
 
-  ``_restart()``, 
-    Seek to the start of ``auxfile``
+  ``_restart()``,
+  Seek to the start of ``auxfile``
```
Comment (lilyminium) [inline package/MDAnalysis/auxiliary/__init__.py]:
```suggestion
    Seek to the start of ``auxfile``
```

The spaces removed here were important for the formatting -- see https://mdanalysis--3646.org.readthedocs.build/en/3646/documentation_pages/auxiliary/init.html

**Review by lilyminium — CHANGES_REQUESTED** [review]

**Review by lilyminium — APPROVED** [review]
LGTM, thanks @aditi2906. @richardjgowers do you want to merge once CI is done?

**Review by richardjgowers — APPROVED** [review]

**Pull request merged**